integritybackend/actions.py

```diff
@@ -207,235 +207,286 @@ def create(self, asset_fullpath, jwt_payload, meta):
-                    custom_assertions_path,
-                )
-        except Exception as err:
-            _logger.info(
-                "No custom assertions dictionary found: %s",
-                custom_assertions_path,
+        meta_content = None
+        photographer_id = None
+        with ZipFile(tmp_zip) as zipf:
+            meta_content_path = next(
+                (s for s in zipf.namelist() if s.endswith("-meta-content.json")), None
             )
-        return custom_assertions_dictionary
+            if meta_content_path is None:
+                raise Exception(f"ZIP at {zip_path} has no content metadata file")
+            with zipf.open(meta_content_path) as meta_content_f:
+                meta_content = json.load(meta_content_f)
+                photographer_id = asset_helper.filename_safe(meta_content["private"]["signal"]["sourceName"])
+
+            # Open content ZIP and extract all JPEGs
+            content_zip = next((s for s in zipf.namelist() if s.endswith(".zip")), None)
+            if content_zip is None:
+                raise Exception(f"ZIP at {zip_path} has no content file")
+            file_util.create_dir(tmp_img_dir)
+            with ZipFile(zipf.open(content_zip)) as content_zip_f:
+                for file_path in content_zip_f.namelist():
+                    if os.path.splitext(file_path)[1].lower() in [".jpg", ".jpeg"]:
+                        content_zip_f.extract(file_path, tmp_img_dir)
+
+        # Get list of JPEGs
+        image_filenames = []
+        for filename in os.listdir(tmp_img_dir):
+            if filename.lower().endswith(".jpg") or filename.lower().endswith(".jpeg"):
+                image_filenames.append(filename)
+        
+        # C2PA-inject all JPEGs
+        for filename in image_filenames:
+            claim = _claim.generate_c2pa_proofmode(action_params, meta_content, filename)
+            path = os.path.join(tmp_img_dir, filename)
+            _claim_tool.run_claim_inject(claim, path, None)
+
+        # Process C2PA-injected JPEGs
+        for filename in image_filenames:
+            # Rename each image file to .jpg
+            path = os.path.join(tmp_img_dir, filename)
+            image_path = FileUtil.change_filename_extension(path, ".jpg")
+            os.rename(path, image_path)
+
+            # Read claims (requires .jpg extension as input)
+            claim_path = FileUtil.change_filename_extension(image_path, ".json")
+            _claim_tool.run_claim_dump(image_path, claim_path)
+
+        # Copy all C2PA-injected JPEGs to action_dir
+        shutil.copytree(tmp_img_dir, action_img_dir, dirs_exist_ok=True)
+
+        # Atomically move all C2PA-injected JPEGs to output folder under photographer ID and date
+        shared_dir = os.path.join(
+            action_output_dir,
+            photographer_id,
+            datetime.now().strftime("%Y-%m-%d"),
+            bundle_name,
```

Comment:
Grouping by SHA256 of ZIP. @YurkoWasHere make sure preproc hands me correctly (sha256) named ZIPs, otherwise the bundles will fail verification above.